cadc-dali-parquet/src/main/java/ca/nrc/cadc/dali/tables/parquet/ParquetReader.java

```diff
@@ -69,199 +69,272 @@
+        return read(parquetInputFile);
+    }
 
-        try (org.apache.parquet.hadoop.ParquetReader<GenericRecord> reader = AvroParquetReader.<GenericRecord>builder(inputFile).build()) {
-            GenericRecord record;
+    /**
+     * Read a Parquet data source via a RandomAccessSource abstraction and produce a VOTableDocument.
+     *
+     * @param randomAccessSource Random-access capable source of Parquet bytes; expected to support seek and bounded reads.
+     * @return VOTableDocument representation of the Parquet content;
+     * @throws IOException if byte access or Parquet parsing fails.
+     */
+    public VOTableDocument read(RandomAccessSource randomAccessSource) throws IOException {
+        log.debug("Reading RandomAccessSource");
+        return read(new RandomSeekableInputFile(randomAccessSource));
+    }
 
-            while ((record = reader.read()) != null) {
-                log.debug("Reading Record: " + record);
-                recordCount++;
+    /**
+     * Read a Parquet stream and produce a VOTableDocument.
+     * It reads the stream fully and stores it in a local temporary file before returning.
+     *
+     * @param inputStream stream of Parquet content.
+     * @return VOTableDocument representation of the Parquet content;
+     * @throws IOException if reading from the stream or parsing the Parquet content fails.
+     */
+    public VOTableDocument read(InputStream inputStream) throws IOException {
+        log.debug("Reading InputStream");
+        File tempFile = null;
+
+        try {
+            String fileName = "parquet-" + UUID.randomUUID();
+            tempFile = File.createTempFile(fileName, ".parquet");
+            log.debug("File " + fileName + " created successfully");
+
+            try (OutputStream out = new FileOutputStream(tempFile)) {
+                MultiBufferIO copier = new MultiBufferIO();
+                copier.copy(inputStream, out);
+            } catch (Exception e) {
+                log.error("Error reading from InputStream", e);
+                throw new RuntimeException(e);
+            }
 
-                // TODO: This is a temporary implementation to read the data
-                Map<String, Object> rowData = new HashMap<>();
-                for (Type field : schema.getFields()) {
-                    String fieldName = field.getName();
-                    rowData.put(fieldName, record.get(fieldName));
+            log.debug("File" + fileName + " prepared successfully for reading");
+
+            RandomAccessFile raf = new RandomAccessFile(tempFile, "r");
+            ParquetInputFile parquetInputFile = new ParquetInputFile(raf);
+            return read(parquetInputFile);
+        } catch (IOException e) {
+            log.error("Failed to read Parquet from InputStream", e);
+            throw new RuntimeException("Error reading Parquet file", e);
+        } finally {
+            try {
+                if (tempFile != null && tempFile.exists()) {
```

Comment:
So the `tempFile.delete()` here relies on it being opened (line 181-183) above so the content will last as long as it is open? 

Is the underlying file ever closed? Does ParquetReader need a close() method (implement Closeable)? 

While it probably relates to the binary2 PR: does VOTableReader also need to be Closeable for the same reason??